### DISCUSSION TITLE
Searching by some arbitrary ID currently stored in the Title

### DISCUSSION THREAD

**Discussion by MrHakisak** [gotson/komga#1723] [Ideas]
[There is a small project](https://github.com/9-FS/nhentai_archivist) where you can scrape certain-manga from a certain-website and we would like to assign an ID number to each one-shot to make it searchable. Currently, the ID number is 1 to 6 digits long and is in the title to make it searchable. It would be nice to search by GTIN or Number, that way we can remove the ID from the title and still make it searchable.

-Thank you

**Comment by gotson** [comment]
you could use the Book metadata Links to store this information. A URI is a perfectly good identifier.

You mention GTIN / ISBN in the title, but you also mention a 1-6 digit ID, which is not the format of a GTIN/ISBN. Komga already has an ISBN field for book metadata.

**Comment by 9-FS** [comment]
Hi gotson, I'm the creator of the aforementioned small project. Thanks for your quick response. :)

I'm currently using CBZ with [ComicInfo.xml](https://anansi-project.github.io/docs/comicinfo/documentation) to save metadata and mainly used your documentation [here](https://komga.org/docs/guides/scan-analysis-refresh/#import-metadata-for-cbrcbz-containing-a-comicinfoxml-file) to understand how Komga interprets it. 

I currently put `"{id} {title}"` in `<Title>` to enable quickly searching by either ID or title via the main search bar. This is not ideal though as it puts 2 types of information into 1 field. My ideal solution would be to put `id` into a separate field (`<Number>`?) **while retaining the ability to search by ID**. 

> you could use the Book metadata Links to store this information

I'm not really sure what you mean by that. I already use the `<Web>` tag to reference the source gallery but I'm not aware how to search by that on Komga.

> You mention GTIN / ISBN in the title, but you also mention a 1-6 digit ID, which is not the format of a GTIN/ISBN. Komga already has an ISBN field for book metadata.

Exactly, which is why I don't use the `<GTIN>` field. Maybe allow arbitrary input there instead validating it?

Please let me know what you think~

**Reply by gotson** [reply]
thanks for the details, however i am a bit lost about what's the actual concern. Maybe if you could explain what the workflow is, and what you expect Komga to provide in order to do "..." ?

It also seem that you use ComicInfo.xml as the main vessel to pass information to Komga, so there may be limitations because ComicInfo.xml's model is more limited that Komga's.

**Reply by 9-FS** [reply]
Okay, let me try to rephrase it.

1. I have many one-shot comics that each have a title and an ID. We're talking about over 100.000 entries.
1. I want to use the main search bar in Komga to search by title or by ID.
1. I currently put `"{id} {title}"` in `<Title>` to achieve this. This is not ideal. I want only `title` in `<Title>` and put `id` in a separate field like `<Number>`, `<GTIN>`, or something else.
1. Currently, searching by `<Number>` is not possible. Making this possible could be a solution.
1. Alternatively, putting `id` in `<GTIN>` does not work, because Komga only accepts it if it is valid and I also don't know if Komga allows searching by `<GTIN>`.  Removing the GTIN validation and enabling searching by it could also be a solution.

**Reply by gotson** [reply]
how do you set the metadata in Komga ? Using the API, or using `ComicInfo.xml` ?

**Reply by 9-FS** [reply]
With `ComicInfo.xml`. The `<Title>`, `<Number>`, and `<GTIN>` i've mentioned previously refer to entries there.

**Reply by gotson** [reply]
You're a bit constrained because you compound 2 things:
1. the search capabilities of Komga
2. the `ComicInfo.xml` and its limitations

To search by arbitrary IDs in Komga, you could store them inside the series alternate titles. But that field is not supported in `ComicInfo.xml`.

**Reply by 9-FS** [reply]
That is not helpful at all... You're ignoring my suggestions and coming up with a solution you yourself say is not suitable for my use-case. I don't have the feeling you're really reading what we're saying.
